app/controllers/OptInController.scala

```diff
@@ -79,9 +82,14 @@ class OptInController @Inject()(
     }
   }
 
+  //if there is a clientList available then proceed to Groups otherwise go back to the ASA dashboard (and wait)
   def showYouHaveOptedIn: Action[AnyContent] = Action.async { implicit request =>
     withAuthorisedAgent { _ =>
-      Future.successful(Ok(you_have_opted_in()))
+      withSession { session =>
+        session.clientList.fold(
+          sessionCacheRepository.deleteFromSession(DATA_KEY).map(_ => Ok(you_have_opted_in(appConfig.agentServicesAccountManageAccountUrl))
+        ))(_ => Ok(you_have_opted_in(routes.GroupsController.root.url)).toFuture)
```

Comment:
Not sure why you added that there. I think it's the backUrl that is passed to that page.

Reply:
this is the complete page so there is no back url - they have just opted in...if the status is READY then they can go onto groups if not then they redirect to ASA dashboard. Some further changes will be required...